### ISSUE TITLE
Foreign key discarded in findOne query

### ISSUE THREAD

**Issue by dimlucas** [typeorm/typeorm#5323]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**
[ ] `0.2.19` 

**Steps to reproduce or a small repository showing the problem:**

I have a model called Shop and a model called Story.
Story has a ManyToOne relationship with Shop defined like this in Story.model.ts:
```ts
@ManyToOne(type => Shop, shop => shop.stories)
@JoinColumn()
shop: Shop;
```
In one of my services where I have injected a Story repository I attempt the following:
```ts
let activeStory = await this._storiesRepository.findOne({
    where: {
        shopId: shopId,
        active: true
   }
}
```
with the intent to get the first active story for the given shop id. However, I always get the first active story in the database without TypeORM ever taking shopId into account.

If I do this:

```ts
let activeStory = await this._storiesRepository.findOne({
    where: {
        shop: {id: shopId},
        active: true
   }
}
```
I get a DeepEquals error. 
The only solution that works is to first fetch the shop and pass it like this:
```ts
let activeStory = await this._storiesRepository.findOne({
    where: {
        shop,
        active: true
   }
}
```
This works and I can see WHERE Shop.id=? in the query.

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
!>

**Comment by mastraub** [comment]
Thanks for sharing. Having the same Issue.

Another way to make
```TypeScript
where: {
        shopId: shopId,
        active: true
   }
```
work, you could add the foreign key column to the entity declaration, like
```TypeScript
@Column()
shopId: number;
```

**Comment by imnotjames** [comment]
This should be fixed by #7805
